CMakeLists.txt

```diff
@@ -24,7 +24,13 @@ else ()
     message(FATAL_ERROR "Compiler ${CMAKE_CXX_COMPILER} has no C++11 support")
 endif ()
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_SYSTEM_NO_DEPRECATED -fopenmp -Wall -Wextra")
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set(MACOSX TRUE)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_SYSTEM_NO_DEPRECATED -DNO_OPENMP -Wall -Wextra")
```

Comment:
Minor thing (no need to do this if you feel there is no need to): move all non-OpenMP-related flags before the `if` (those that will be included regardless if the compiling machine is Linux or Mac), and in the `if` just add `-fopenmp` or `-DNO_OPENMP`. This is to avoid us to have to consistently change both lines when compiling flags change.

Reply:
Agreed. Done.